### ISSUE TITLE
Node dead after start [ DELTA 6 ]

### ISSUE THREAD

**Issue by matsuro-hadouken** [CasperLabs/casper-node#698]
Trying to join delta-6 recently, after couple of minutes node crash with provided error:

```
   0: casper_node::panic_hook
   1: core::ops::function::Fn::call
   2: std::panicking::rust_panic_with_hook
             at rustc/25f6938da459a57b43bdf16ed6bdad3225b2a3ce/library/std/src/panicking.rs:597:17
   3: std::panicking::begin_panic_handler::{{closure}}
             at rustc/25f6938da459a57b43bdf16ed6bdad3225b2a3ce/library/std/src/panicking.rs:499:13
   4: std::sys_common::backtrace::__rust_end_short_backtrace
             at rustc/25f6938da459a57b43bdf16ed6bdad3225b2a3ce/library/std/src/sys_common/backtrace.rs:141:18
   5: rust_begin_unwind
             at rustc/25f6938da459a57b43bdf16ed6bdad3225b2a3ce/library/std/src/panicking.rs:495:5
   6: std::panicking::begin_panic_fmt
             at rustc/25f6938da459a57b43bdf16ed6bdad3225b2a3ce/library/std/src/panicking.rs:437:5
   7: casper_node::components::consensus::<impl casper_node::components::Component<REv> for casper_node::components::consensus::era_supervisor::EraSupervisor<I>>::handle_event
   8: <casper_node::reactor::validator::Reactor as casper_node::reactor::Reactor>::dispatch_event
   9: casper_node::cli::Cli::run::{{closure}}
  10: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  11: std::thread::local::LocalKey<T>::with
  12: tokio::runtime::enter::Enter::block_on
  13: tokio::runtime::thread_pool::ThreadPool::block_on
  14: tokio::runtime::context::enter
  15: casper_node::main
  16: std::sys_common::backtrace::__rust_begin_short_backtrace
  17: std::rt::lang_start::{{closure}}
  18: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
             at rustc/25f6938da459a57b43bdf16ed6bdad3225b2a3ce/library/core/src/ops/function.rs:259:13
      std::panicking::try::do_call
             at rustc/25f6938da459a57b43bdf16ed6bdad3225b2a3ce/library/std/src/panicking.rs:381:40
      std::panicking::try
             at rustc/25f6938da459a57b43bdf16ed6bdad3225b2a3ce/library/std/src/panicking.rs:345:19
      std::panic::catch_unwind
             at rustc/25f6938da459a57b43bdf16ed6bdad3225b2a3ce/library/std/src/panic.rs:396:14
      std::rt::lang_start_internal
             at rustc/25f6938da459a57b43bdf16ed6bdad3225b2a3ce/library/std/src/rt.rs:51:25
  19: main
  20: __libc_start_main
  21: _start

panicked at 'fatal error [node/src/components/consensus/era_supervisor.rs:811]: Consensus shutting down due to inability to participate in the network', node/src/effect.rs:423:9
```
* Update

Probably getting this from nodes which are on previous delta still.

```
{
  "timestamp": "Dec 30 09:32:32.937",
  "level": "INFO",
  "fields": {
    "message": "NodeId::Tls(3da7..bf59): dropping connection to NodeId::Tls(4037..0b1e) due to genesis config hash mismatch",
    "our_hash": "42edd6b09c101a65d7aa3626bf0811cb39fa4d2d3957d8a28868edd4aa8d1a9e",
    "their_hash": "0e9f4c541abeb97a34708f732b6d0acc7345dbda394ae115ffef213461d1a892"
  },
```

* Update 2

Attached logs are in pure JSON format, can be easily to operate, this is full cycle from start to crash.

_Inside zip is tar.xz archive, GitHub only support zip, sorry for it_

[delta-6.zip](https://github.com/CasperLabs/casper-node/files/5753779/delta-6.zip)

**Comment by Fraser999** [comment]
Thanks for this report too :)

It seems that this is an effect of increasing the `gossip_interval` in the config to `300_000` (i.e. 5 minutes).  We overlooked the fact that the node's consensus component requires there to have been some consensus messages within 5 minutes of the component being constructed or else it deems itself non-functioning and kills the node.

To receive consensus messages, the node must have some incoming connections (the peers listed via the `/status` endpoint could be incoming or outgoing ones).  To get an incoming connection, the node must gossip its public address, and remote peers receiving this will then try and connect to it.

By only having peers gossip their addresses on a 5 minute interval, there's no chance for a node joining the network to establish incoming connections in time to satisfy the consensus requirement.

The simplest fix (which worked when we tried on one of our own nodes) is to reduce the config setting back down.  We opted to change it to:
```
gossip_interval = 120_000
```

Manually making that config change and restarting the node should allow it to join successfully.

I'll leave this issue open until we fix that setting in our release packages.
